docs/develop/web-client-performance.md

```diff
@@ -0,0 +1,107 @@
+
+```TypeScript
+/**
+ * This script provides the count of negative numbers that are present
+ * in the used range of the current worksheet.
+ */
+function main(workbook: ExcelScript.Workbook) {
+  // Get the working range.
+  let usedRange = workbook.getActiveWorksheet().getUsedRange();
+
+  // Save the values locally to avoid repeatedly asking the workbook.
+  let usedRangeValues = usedRange.getValues();
+  let rowCount = usedRange.getRowCount();
+  let columnCount = usedRange.getColumnCount();
+
+  // Start the negative number counter.
+  let negativeCount = 0;
+
+  // Iterate over the entire range looking for negative numbers.
+  for (let i = 0; i < rowCount; i++) {
+    for (let j = 0; j < columnCount; j++) {
+      if (usedRangeValues[i][j] < 0) {
+        negativeCount++;
+      }
+    }
+  }
+
+  // Log the negative number count to the console.
+  console.log(negativeCount);
+}
+```
+
+> [!NOTE]
+> As an experiment, try replacing `usedRangeValues`, `rowCount`, and `columnCount` in the loop with the respective `get` methods. You may notice the script takes considerably longer to run when dealing with large ranges.
+
+### Remove unnecessary `console.log` statements
+
+Console logging is a vital tool for [debugging your scripts](../testing/troubleshooting.md). However, it does force the script to synchronize with the workbook to ensure the logged information is up-to-date. This typically won't cause a noticeable performance issue, but you should consider removing unnecessary logging statements (such as those used for testing) before sharing your script.
+
+### Avoid using try/catch blocks
+
+We don't recommend using [`try`/`catch` blocks](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Statements/try...catch) in an Office Script. Most errors can be avoided by checking objects returned from the workbook. For example, the following script checks that the table returned by the workbook exists before trying to add a row.
+
+```TypeScript
+/**
+ * This script adds a row to "MyTable", if that table is present.
+ */
+function main(workbook: ExcelScript.Workbook) {
+  let table = workbook.getTable("MyTable");
+
+  // Check if the table exists.
+  if (table) {
+    // Add the row.
+    table.addRow(-1, ["2012", "Yes", "Maybe"]);
+  } else {
+    // Report the missing table.
+    console.log("MyTable not found.");
+  }
+}
+```
+
+## Control the workbook-script synchronization
+
+If your script has a critical performance need that is not solved by applying these techniques, you can manually control the workbook synchronization. Use the Office Scripts Async APIs to do this. These APIs let you batch together read and write calls in a way that is most optimal for your script. [Here's a primer on the Office Scripts Async APIs](excel-async-model.md).
```

Comment:
Could we add a contact us piece? We are very interested in understanding what performance problems people are running into.

Reply:
There's a feedback control there for the docs, but I'll add a special piece directing people to Stack Overflow.